pkg/cmd/cli/cmd/exporter.go

```diff
@@ -0,0 +1,216 @@
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/registry/controller"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/registry/endpoint"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/registry/minion"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/registry/namespace"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/registry/persistentvolume"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/registry/persistentvolumeclaim"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/registry/pod"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/registry/resourcequota"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/registry/secret"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/registry/serviceaccount"
+
+	buildapi "github.com/openshift/origin/pkg/build/api"
+	buildrest "github.com/openshift/origin/pkg/build/registry/build"
+	buildconfigrest "github.com/openshift/origin/pkg/build/registry/buildconfig"
+	deployapi "github.com/openshift/origin/pkg/deploy/api"
+	imageapi "github.com/openshift/origin/pkg/image/api"
+	routeapi "github.com/openshift/origin/pkg/route/api"
+)
+
+var ErrExportOmit = fmt.Errorf("object is omitted")
+
+type Exporter interface {
+	AddExportOptions(*pflag.FlagSet)
+	Export(obj runtime.Object, exact bool) error
+}
+
+type defaultExporter struct{}
+
+func (e *defaultExporter) AddExportOptions(flags *pflag.FlagSet) {
+}
+
+func exportObjectMeta(objMeta *kapi.ObjectMeta, exact bool) {
+	objMeta.UID = ""
+	if !exact {
+		objMeta.Namespace = ""
+	}
+	objMeta.CreationTimestamp = util.Time{}
+	objMeta.DeletionTimestamp = nil
+	objMeta.ResourceVersion = ""
+	objMeta.SelfLink = ""
+	if len(objMeta.GenerateName) > 0 && !exact {
+		objMeta.Name = ""
+	}
+}
+
+func (e *defaultExporter) Export(obj runtime.Object, exact bool) error {
+	if meta, err := kapi.ObjectMetaFor(obj); err == nil {
+		exportObjectMeta(meta, exact)
+	} else {
+		glog.V(4).Infof("Object of type %v does not have ObjectMeta: %v", reflect.TypeOf(obj), err)
+	}
+	switch t := obj.(type) {
+	case *kapi.Endpoints:
+		endpoint.Strategy.PrepareForCreate(obj)
+	case *kapi.ResourceQuota:
+		resourcequota.Strategy.PrepareForCreate(obj)
+	case *kapi.LimitRange:
+	// TODO: this needs to be fixed
+	//  limitrange.Strategy.PrepareForCreate(obj)
+	case *kapi.Node:
+		minion.Strategy.PrepareForCreate(obj)
+		if exact {
+			return nil
+		}
+		t.Status = kapi.NodeStatus{}
+	case *kapi.Namespace:
+		namespace.Strategy.PrepareForCreate(obj)
+	case *kapi.PersistentVolumeClaim:
+		persistentvolumeclaim.Strategy.PrepareForCreate(obj)
+	case *kapi.PersistentVolume:
+		persistentvolume.Strategy.PrepareForCreate(obj)
+	case *kapi.ReplicationController:
+		controller.Strategy.PrepareForCreate(obj)
+	case *kapi.Pod:
+		pod.Strategy.PrepareForCreate(obj)
+	case *kapi.PodTemplate:
+	case *kapi.Service:
+		// TODO: service does not yet have a strategy
+		t.Status = kapi.ServiceStatus{}
```

Comment:
Nodes are the only resource that have editable status by admins (somewhat arbitrary).  Will add a comment.

----- Original Message -----
> > +		}
> > +		t.Status = kapi.NodeStatus{}
> > +	case *kapi.Namespace:
> > +		namespace.Strategy.PrepareForCreate(obj)
> > +	case *kapi.PersistentVolumeClaim:
> > +		persistentvolumeclaim.Strategy.PrepareForCreate(obj)
> > +	case *kapi.PersistentVolume:
> > +		persistentvolume.Strategy.PrepareForCreate(obj)
> > +	case *kapi.ReplicationController:
> > +		controller.Strategy.PrepareForCreate(obj)
> > +	case *kapi.Pod:
> > +		pod.Strategy.PrepareForCreate(obj)
> > +	case *kapi.PodTemplate:
> > +	case *kapi.Service:
> > +		// TODO: service does not yet have a strategy
> > +		t.Status = kapi.ServiceStatus{}
> 
> Not sure about this happening before checking for `--exact`. You preserve
> status in nodes but not here. Why?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/openshift/origin/pull/2910/files#r31895569